### PR TITLE
add security token to profile 

### PR DESCRIPTION
**Pull request by fogfish**
This is improvement of previous PR
https://github.com/erlcloud/erlcloud/pull/293

### CODE REVIEW

File: src/erlcloud_aws.erl [hunk 1]
```diff
@@ -769,34 +769,43 @@ profiles_resolve( Name, Profiles, Role, Options ) ->
...
     BinRole = proplists:get_value( role_arn, Keys ),
     case [proplists:get_value( K, Keys ) || K <- Names] of
-        [Id, Secret, undefined] when Id =/= undefined, Secret =/= undefined ->
-            {ok, binary_to_list(Id), binary_to_list(Secret)};
-        [undefined, undefined, BinProfile] when BinProfile =/= undefined ->
+        [Id, Secret, undefined, undefined] when Id =/= undefined, Secret =/= undefined ->
+            {ok, {binary_to_list(Id), binary_to_list(Secret)}};
+        [Id, Secret, Token, undefined] when Id =/= undefined, Secret =/= undefined, Token =/= undefined ->
+            {ok, {binary_to_list(Id), binary_to_list(Secret), binary_to_list(Token)}};
+        [undefined, undefined, _, BinProfile] when BinProfile =/= undefined ->
```
Comment (motobob) [inline src/erlcloud_aws.erl]: may be you meant?
```erlang
[undefined, undefined, undefined, BinProfile]
```

Comment (fogfish) [inline src/erlcloud_aws.erl]: I thought about it. Frankly speaking I do not know how profile works with security token. Maybe you can both security token and profile... 

having this, you will crush the code if security toke is defined for profile.
```
[undefined, undefined, undefined, BinProfile]
```

Comment (fogfish) [inline src/erlcloud_aws.erl]: So what is our conclusion?
```
[undefined, undefined, undefined, BinProfile]
or
[undefined, undefined, _, BinProfile]
```

Comment (motobob) [inline src/erlcloud_aws.erl]: i would prefer to be as close as possible to AWS CLI, where together with source profile user sets only the role.
 so `[undefined, undefined, undefined, BinProfile]`

**Comment by fogfish** [comment]
right, fixed based on comment

**Pull request merged**